agent/plugin/error.go

```diff
@@ -0,0 +1,123 @@
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+type unit struct{}
+
+// DeprecatedNameErrors contains a set of DeprecatedNameError
+type DeprecatedNameErrors struct {
+	errs map[DeprecatedNameError]unit
+}
+
+// IsEmpty return true if and only if `e` contains no errors
+func (e *DeprecatedNameErrors) IsEmpty() bool {
+	return e == nil || len(e.errs) == 0
+}
+
+// Errors returns the contained set of errors in sorted order
+func (e *DeprecatedNameErrors) Errors() []DeprecatedNameError {
+	if e == nil {
+		return []DeprecatedNameError{}
+	}
+
+	errs := make([]DeprecatedNameError, 0, len(e.errs))
+	for err := range e.errs {
+		errs = append(errs, err)
+	}
+
+	sort.Slice(errs, func(i, j int) bool {
+		if errs[i].old == errs[j].old {
+			return errs[i].new < errs[j].new
+		}
+		return errs[i].old < errs[j].old
+	})
+
+	return errs
+}
+
+// Error returns each contained error on a new line
+func (e *DeprecatedNameErrors) Error() string {
+	builder := strings.Builder{}
+	for i, err := range e.Errors() {
+		_, _ = builder.WriteString(err.Error())
```

Comment:
Builder.WriteString (and WriteRune) are [documented](https://pkg.go.dev/strings#Builder.WriteString) as returning nil error always, so the underscores seem extra unnecessary here.

Reply:
I don't think it's too distracting to leave it this way. It makes the life of static analysis tools easier.